Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -117,6 +117,7 @@ def __init__(self, session, tdef):
         self.vod_index = None
         self.orig_files = None
         self.finished_callback = None
+        self.finished_callback_already_called = False
```

Comment:
I wonder if this will have unexpected consequences, since the finished alert raised by libtorrent is not always predictable.

Maybe it's better to handle this error at a higher level (when processing the channel blobs)? Is there a way to not process the same data again, while allowing `finished_callback` to fire multiple times?

Reply:
I've already added the code to skip blobs based on current torrent version, so it is relatively safe to remove this. However, it would still be dangerous to try to process the dir multiple times.

But what do you mean by "unpredictable"? Do you mean that it can trigger prematurely, not just repeatedly?

Reply:
Ok, I'll leave it out. The checks at the `process_dir` level should be enough.

Reply:
The warning is still there, though.